packages/protocol/contracts/governance/BlockchainParameters.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../common/Initializable.sol";
+
+/**
+ * @title Contract for storing blockchain parameters that can be set by governance.
+ */
+contract BlockchainParameters is Ownable, Initializable {
+
+  struct ClientVersion {
+    uint256 major;
+    uint256 minor;
+    uint256 patch;
+  }
+
+  ClientVersion minimumClientVersion;
+
+  event MinimumClientVersionSet(uint256 major, uint256 minor, uint256 patch);
+
+  /**
+   * @notice Initializes critical variables.
+   * @param _minimumClientVersion1 Minimum client version that can be used in the chain,
+   * major version.
+   * @param _minimumClientVersion2 Minimum client version that can be used in the chain,
+   * minor version.
+   * @param _minimumClientVersion3 Minimum client version that can be used in the chain,
+   * patch level.
+   */
+  function initialize(
+    uint256 _minimumClientVersion1,
```

Comment:
Naming: include `major`, `minor`, etc. in these names, not numbers. No need for `_`s if the names are not shadowing storage variables.